### PR TITLE
Add resources to Metroid DNA actor pickups

### LINKED ISSUE #143: metroid DNA pickup actors don't activate metroidnization properly

### PR DESCRIPTION
**Pull request by MayberryZoom**
As titled, adds resources to Metroid DNA actor pickups. Fixes #143.

I'm opening this as a draft PR to get feedback, partially because I've done a bit of refactoring. Since I'm using what is currently the progressive template, I generalized it into a custom powerup template. This only renames the template itself and a few variable/property names in the Lua editor.

The other reason this is a draft is because I'm having an issue that I can't figure out. With my code changes, picking up DNA from an actor grants a second DNA (seemingly without calling `OnPickedUp()`) after `OnPickedUp()` finishes. This doesn't happen with other pickups, only with DNA. Advice on this would be highly appreciated.

### CODE REVIEW

**Review by duncathan — CHANGES_REQUESTED** [review]
the double item is probably because the actor is granting the item as well, instead of granting `ITEM_NONE` like progressives do

File: src/open_dread_rando/pickups/lua_editor.py [hunk 1]
```diff
@@ -52,11 +52,37 @@ def get_parent_for(self, item_id) -> str:
 
     def get_script_class(self, pickup: dict, boss: bool = False, actordef_name: str = "") -> str:
         pickup_resources = pickup["resources"]
-        parent = self.get_parent_for(pickup_resources[0][0]["item_id"])
+        parent = pickup_resources[0][0]
```
[the remark this answers is not in the document]
Comment (MayberryZoom) [reply src/open_dread_rando/pickups/lua_editor.py:55]: What is it then, if not the parent?

[the remark this answers is not in the document]
Comment (duncathan) [reply src/open_dread_rando/pickups/lua_editor.py:55]: a single resource, in this case

**Comment by MayberryZoom** [comment]
Ready for review; double item issue fixed.

File: src/open_dread_rando/pickups/lua_editor.py [hunk 2]
```diff
@@ -52,11 +52,37 @@ def get_parent_for(self, item_id) -> str:
 
     def get_script_class(self, pickup: dict, boss: bool = False, actordef_name: str = "") -> str:
         pickup_resources = pickup["resources"]
-        parent = self.get_parent_for(pickup_resources[0][0]["item_id"])
+        first_resource = pickup_resources[0][0]
 
         if not boss and len(pickup_resources) == 1 and len(pickup_resources[0]) == 1:
+            parent_id = first_resource["item_id"]
```
Comment (duncathan) [inline src/open_dread_rando/pickups/lua_editor.py:58]: nope lol

Comment (MayberryZoom) [reply src/open_dread_rando/pickups/lua_editor.py:58]: Just tell me what to name it lol

**Review by duncathan — CHANGES_REQUESTED** [review]
please fix the ruff errors. you probably don't have pre-commit installed correctly

**Review by duncathan — APPROVED** [review]

**Pull request merged**